### ISSUE TITLE
You can accidentally delete most of your observations by logging out and back in again

### ISSUE THREAD

**Issue by kaldari** [inaturalist/INaturalistIOS#499]
Steps to reproduce:
1. Log out of the iOS app
2. Log back into the iOS app
3. Go to the "Me" tab (quickly)
4. Either click on the sync button or pull down on the list (quickly)

Expected result: Nothing special happens, it just continues to download your observations to the phone until they're all there.

Actual result: It sees that the list of observations on your phone is different from the list of observations on the website (since it hasn't downloaded them all to the phone yet), so it deletes all the observations on the website that weren't yet on your phone (under the assumption that you manually deleted them).

It seems like this can cause you to accidentally delete most (or all) of your observations from iNat.

**Comment by kaldari** [comment]
@kueda - Yes, I accidentally deleted a couple hundred of my observations this way :(

**Comment by tiwane** [comment]
I *think* I was able to accidentally replicate this bug today, or at least sort of replicate it. I was checking how changing one’s common names place affected the iOS app (for a help@inat email) so I changed my common names place to Germany on the website. I believe I had signed out of the app but I can’t be sure. When I signed back in, it took forever to load my obs on the Me tab. I finally pressed “cancel” and the button turned into “Deletes to sync” - no obs showed up, though. I can't remember if I pulled down to try and refresh again. When I signed out and signed back in again, everything looked fine, but when I checked some other observations today, they're gone. 

@kaldari did you make any changes to your account before signing back in?

**Comment by kaldari** [comment]
No, I signed back in right after signing out. I also remember seeing the "Deletes to sync" button. I'm pretty sure I didn't click the button, but pulled down to try to refresh the list instead.

**Comment by kaldari** [comment]
Aside from making sure any downloads have finished before allowing syncing from the app, it would also be a good idea to have the user confirm their intentions when syncing: "Are you sure you want to delete 200 observations?" That way even if there is a bug, it won't cause as much harm.

**Comment by alexshepard** [comment]
Ugh. We've run into this before, and I thought I had fixed it.

Thanks for reporting @kaldari .

I think this is a problem with the way we handle DeletedRecords in the case where the user signing out, and potentially a database race condition between signing out and observation fetch. Will try to investigate today and get a fix out this week.

**Comment by alexshepard** [comment]
A few fixes for this were made in release 2.7.18, released on 24 April. I'm going to keep this open in case it recurs, but I'm going to remove the 2.7.18 milestone.
